src/sage/numerical/pivot_rules_for_simplex_method.py

```diff
@@ -0,0 +1,776 @@
+    See [BDLS2022]_.
+
+    EXAMPLES::
+
+        sage: from sage.numerical.pivot_rules_for_simplex_method import SimplexMethodPivot
+        sage: A = ([1, 1], [3, 1])
+        sage: b = (1000, 1500)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: D = P.initial_dictionary()
+        sage: pivot = SimplexMethodPivot("NW_rule")
+        sage: def eta(v): # use 2 norm as normalization
+        ....:     return v.norm()
+        sage: pivot(D, False, eta, [1,2])
+        sage: D.entering()
+        x2
+        sage: D.leaving()
+        x3
+
+    NW rule is not eqipped to solve both a Phase I and a phase II with :meth:`run_simplex_method`::
+
+        sage: A = ([1, 1], [3, 1], [-1, -1])
+        sage: b = (1000, 1500, -400)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: def eta(v):
+        ....:     return v.norm()
+        sage: P.run_simplex_method('NW_rule', eta, [1,2])
+        Traceback (most recent call last):
+        ...
+        TypeError: unsupported operand parent(s) for *: 'Ambient free module of rank 2 over the principal ideal domain Integer Ring' and 'Vector space of dimension 3 over Rational Field'
+
+    A Phase I and a Prase II problem require different NW rules due to the difference in ambient dimension of the respective problems::
+
+        sage: A = ([1, 1], [3, 1], [-1, -1])
+        sage: b = (1000, 1500, -400)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: def eta(v):
+        ....:     return v.norm()
+        sage: D = P.auxiliary_problem().initial_dictionary() # Phase I
+        sage: D.enter('x0')
+        sage: D.leave('x5')
+        sage: D.update()
+        sage: D.run_simplex_method('NW_rule', eta, [1,1,4])
+        \begin{equation*}
+        ...
+        \end{equation*}
+        Entering: $x_{2}$. Leaving: $x_{0}$.
+        \begin{equation*}
+        ...
+        \end{equation*}
+        sage: d = P.feasible_dictionary(D)
+        sage: d.run_simplex_method('NW_rule', eta, [1,1])
+        \begin{equation*}
+        ...
+        \end{equation*}
+        Entering: $x_{5}$. Leaving: $x_{3}$.
+        \begin{equation*}
+        ...
+        \end{equation*}
+        Entering: $x_{1}$. Leaving: $x_{4}$.
+        \begin{equation*}
+        ...
+        \end{equation*}
+    """
+    def dictionary_pivot(current_dictionary, eta, weight):
+        r"""
+        Sets the entering and leaving variables of `:class:`~sage.numerical.interactive_simplex_method.LPDictionary` as defined by Normalized Weight pivot rule.
+
+        Input:
+        - ``current_dictionary`` -- an instance of `:class:`~sage.numerical.interactive_simplex_method.LPDictionary`
+
+        - ``eta`` -- callable function from ambient space of ``current_dictionary`` to the real numbers.
```

Comment:
I think it would be better if an instance of the class `NWRule` had eta and weight.
After all, as you write, "an NW-rule is defined by a normalization"